test/jdk/java/lang/Thread/virtual/MonitorWaitNotify.java

```diff
@@ -640,6 +640,60 @@ void testParkingPermitNotOffered() throws Exception {
         assertTrue(completed.get());
     }
 
+    /**
+     * Test no deadlock happens when Object.wait is called from a mix of pinned and non-pinned
+     * paths and notification is done using notifyAll.
```

Comment:
At some point then maybe we should combine this with RetryMonitorEnterWhenPinned. I'm not suggesting we do this now but some of the expanded description might be useful to include here as a passing reader might not immediately know what this test is doing.

Reply:
I wasn’t sure where to put the extra test and missed `RetryMonitorEnterWhenPinned.java`. I agree it makes more sense to have it here. Moved now, let me know what you think.